circus-train-aws/src/main/java/com/hotels/bdp/circustrain/aws/S3DataManipulator.java

```diff
@@ -54,6 +54,11 @@ public boolean delete(String path) {
     List<KeyVersion> keysToDelete = getKeysToDelete(bucket, uri.getKey());
     log.debug("Deleting keys: {}", keysToDelete.stream().map(k -> k.getKey()).collect(Collectors.toList()));
 
+    if (keysToDelete.isEmpty()) {
+      log.info("Nothing to delete at location: {}", path);
+      return false;
```

Comment:
Is `false` the most appropriate return value here? Although this is a special case, we have that 0 files have been deleted out of 0 requested (unless the return value is also an indication that a `delete` operation did actually take place).

Reply:
Hmm, I'm not actually sure, I couldn't find the result of this method used anywhere

Reply:
Same here, the only call I have found for this method discards the return value. @patduin, what do you think?

Reply:
I think it's ok, false is what `io.File.delete()` does.

Unfortunately this class does not have a test, luckily that will soon be fixed right @andreeapad ?!? :)

Reply:
yes, good point @patduin  😄